org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/logging/LoggingTester.java

```diff
@@ -0,0 +1,352 @@
+			return result;
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj)
+				return true;
+			if (obj == null)
+				return false;
+			if (getClass() != obj.getClass())
+				return false;
+			LogEntry other = (LogEntry) obj;
+			if (level == null) {
+				if (other.level != null)
+					return false;
+			} else if (!level.equals(other.level))
+				return false;
+			if (message == null) {
+				if (other.message != null)
+					return false;
+			} else if (!message.equals(other.message))
+				return false;
+			if (source == null) {
+				if (other.source != null)
+					return false;
+			} else if (!source.equals(other.source))
+				return false;
+			if (timeStamp != other.timeStamp)
+				return false;
+			return true;
+		}
+
+		@Override
+		public String toString() {
+			ToStringBuilder b = new ToStringBuilder(this);
+			b.add("message", message);
+			b.add("source", source);
+			b.add("timeStamp", timeStamp);
+			b.add("level", level);
+			return b.toString();
+		}
+
+		public String getMessage() {
+			return message;
+		}
+
+		public String getSource() {
+			return source;
+		}
+
+		public long getTimeStamp() {
+			return timeStamp;
+		}
+
+		public Level getLevel() {
+			return level;
+		}
+	}
+
+	private static class QueueAppender extends AppenderSkeleton {
+		private final Queue<LoggingTester.LogEntry> events = new ConcurrentLinkedQueue<LoggingTester.LogEntry>();
+
+		@Override
+		public boolean requiresLayout() {
+			return false;
+		}
+
+		@Override
+		public void close() {
+		}
+
+		@Override
+		protected void append(LoggingEvent event) {
+			LoggingTester.LogEntry entry = new LoggingTester.LogEntry(event.getRenderedMessage(), event.getLoggerName(),
+					event.getTimeStamp(), event.getLevel());
+			events.add(entry);
+		}
+
+		@SuppressWarnings("unused")
+		public Queue<LoggingTester.LogEntry> getEvents() {
+			return events;
+		}
+	}
+
+	private static class SourceFilter extends Filter {
+		private final Logger source;
+
+		@Override
+		public int decide(LoggingEvent event) {
+			if (Objects.equal(event.getLoggerName(), source.getName())) {
+				return Filter.DENY;
+			} else {
+				return Filter.NEUTRAL;
+			}
+		}
+
+		public SourceFilter(Logger source) {
+			this.source = source;
+		}
+
+		@Override
+		public int hashCode() {
+			final int prime = 31;
+			int result = 1;
+			result = prime * result + ((source == null) ? 0 : source.hashCode());
+			return result;
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj)
+				return true;
+			if (obj == null)
+				return false;
+			if (getClass() != obj.getClass())
+				return false;
+			SourceFilter other = (SourceFilter) obj;
+			if (source == null) {
+				if (other.source != null)
+					return false;
+			} else if (!source.equals(other.source))
+				return false;
+			return true;
+		}
+
+		@Override
+		public String toString() {
+			return new ToStringBuilder(this).addAllFields().toString();
+		}
+
+		@SuppressWarnings("unused")
```

Comment:
Better remove this method then to supress the warning!?

Reply:
dont want to refactor deprecated code